test/blackbox-tests/file-extensions.t

```diff
@@ -0,0 +1,86 @@
+Demonstrate different file extensions
+
+  $ . ./setup.sh
+  $ cat > x.ml <<EOF
+  > let () = Js.log "hello"
+  > EOF
+
+Specify the output filename with `-o`
+
+  $ melc -impl x.ml -o x.js
+  $ cat x.js
+  // Generated by Melange
+  'use strict';
+  
+  
+  console.log("hello");
+  
+  /*  Not a pure module */
+
+Try other extensions
+
+  $ melc -impl x.ml -o x.bs.js
+  $ cat x.bs.js
+  // Generated by Melange
+  'use strict';
+  
+  
+  console.log("hello");
+  
+  /*  Not a pure module */
+
+  $ melc -impl x.ml -o x.ml.js
+  $ cat x.ml.js
+  // Generated by Melange
+  'use strict';
+  
+  
+  console.log("hello");
+  
+  /*  Not a pure module */
+
+  $ melc -impl x.ml -o x.melange.js
+  $ cat x.melange.js
+  // Generated by Melange
+  'use strict';
+  
+  
+  console.log("hello");
+  
+  /*  Not a pure module */
+
+  $ rm *
+  $ cat > x.ml <<EOF
+  > let () = Js.log "hello"
+  > EOF
+  $ melc -impl x.ml -o x.foo.
```

Comment:
this is weird. we should check that the extension doesn't start / end with `.`.

Reply:
There is some function `Ext_filename.get_all_extensions_maybe` that runs before the `Ext_js_suffix` functions has a chance to kick in:

https://github.com/melange-re/melange/blob/dc6723376ae99212440b98d8007162eb4db3ce9a/bin/melc.ml#L309

Should we get rid of it or adapt it to ignore extensions ending with `.`?

Reply:
probably a good idea to adapt, perhaps in a follow-up PR